dts/bindings/gpio/rohm,bd8lb600fs.yaml

```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2022 SILA Embedded Solutions GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: |
+    This is a representation of the Rohm BD8LB600FS SPI Gpio Expander.
+
+compatible: "rohm,bd8lb600fs"
+
+include: [gpio-controller.yaml, spi-device.yaml]
+
+properties:
+  "#gpio-cells":
+    const: 2
+
+  ngpios:
+    type: int
+    required: true
+    const: 8
```

Comment:
I think this should not be const. It should be set at the device tree level and not at the binding level.

Reply:
I went ahead and checked the current state of the ngpios property of other GPIO controllers. Interestingely, there are a bunch of other bindings which also define this property as const:

- cypress,cy8c95xx-gpio-port
- microchip,mcp23s17
- nxp,pcal6408a
- nxp,pcf8574
- semtech,sx1509b
- ti,sn74hc595
- ti,tca9538

And then there is the case of these, which define it as enum, hence limited to a set of values:

- microchip,mcp23sxx
- microchip,mcp230xx

All these bindings have in common, that they are basically GPIO expanders. One is an 8 bit shift register, but for the sake of this argument I think this doesn't make a huge difference.

Going back to the description of ngpios in the GPIO controller I think it is correct ot define this property as const for an external chip:

> This property indicates the number of in-use slots of available slots
> for GPIOs. The typical example is something like this: the hardware
> register is 32 bits wide, but only 18 of the bits have a physical
> counterpart. The driver is generally written so that all 32 bits can
> be used, but the IP block is reused in a lot of designs, some using
> all 32 bits, some using 18 and some using 12. In this case, setting
> "ngpios = <18>;" informs the driver that only the first 18 GPIOs, at
> local offset 0 .. 17, are in use.  For cases in which there might be
> holes in the slot range, this value should be the max slot number-1.

Considering this it makes sense to keep this property as not const for all the MCU internal GPIO controllers, as in these cases it really will differ on the actual type of the MCU how many GPIOs the driver will have to support, as the driver will be reused for multiple MCUs from the same family. In this case, as this driver and binding are specific for exactly one chip which has a constant number of GPIOs I think I should stick consequently to how this property is defined for other GPIO expanders.

Reply:
Please see my reply to a similar concern raised by @cfriedt here: https://github.com/zephyrproject-rtos/zephyr/pull/55093#discussion_r1118462208

Reply:
> Considering this it makes sense to keep this property as not const for all the MCU internal GPIO controllers, as in these cases it really will differ on the actual type of the MCU how many GPIOs the driver will have to support, as the driver will be reused for multiple MCUs from the same family. In this case, as this driver and binding are specific for exactly one chip which has a constant number of GPIOs I think I should stick consequently to how this property is defined for other GPIO expanders.

Agreed.